test/checks/mobile/target-size.js

```diff
@@ -177,6 +177,16 @@ describe('target-size tests', function () {
           '#obscurer2'
         ]);
       });
+
+      it('returns true if the obscuring widget is a descendant', function () {
+        var checkArgs = checkSetup(
+          `<a role="link" aria-label="play" tabindex="0" style="display:inline-block" id="target">
+            <button tabindex="-1" style="margin:1px; line-height:20px">Play</button>
```

Comment:
If the button were to not have `tabindex=-1` (so failing the `nested-interactive-role` check), would we still want to fail this rule? Should this rule instead ignore descendant widgets of the widget?

Reply:
That's where I started too, but you can correctly nest controls. Links in particular can contain other controls. I don't think we can ignore those. It could be used in cards for example, where the card is a link, and then inside it there are more controls.

I don't think we can outright ignore nested controls. That'll miss things it shouldn't. There is a potential for false positives on this, I don't know if those are going to happen in practice though. I've only ever seen this nested control pattern with the inner one with tabindex="-1". We could report it as incomplete, but I suspect it's not going to be necessary.

Reply:
I feel that this is too broad of a solution. This only checks if the element is not tabbable, so non-descendant elements would be ignored as well. I think if we should also check that the element is a descendant of the element it's obscuring before we ignore it.

```html
<div style="position: relative">
  <button>hello</button>
  <button tabindex="-1" style="position: absolute; left: 0; top: 0>world</button>
</div>
```